overlord/ifacestate/ifacestate.go

```diff
@@ -134,8 +134,9 @@ func checkConnectConflicts(st *state.State, plugSnap, slotSnap string, auto bool
 }
 
 // Connect returns a set of tasks for connecting an interface.
-//
 func Connect(st *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
+	plugSnap, plugName, slotSnap, slotName = remapIncomingConnStrings(st, plugSnap, plugName, slotSnap, slotName)
```

Comment:
this merits a comment, afaict we do it mostly because of the immediate conflict check?

Reply:
Yes, that's correct. I will add a comment.